rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -378,7 +378,7 @@ def __init__(self) -> None:
         super().__init__()
         rasa.shared.utils.io.raise_warning(
```

Comment:
can you use `raise_deprecation_warning` instead?

Reply:
sure, also changing it to `FutureWarning` . `DeprecationWarning` doesn't make sense here.

Reply:
why? I think that it has implication on how it is displayed to users. I wouldn't change this without looking at the impact on UX, and I think it's beyond the scope of the PR 🤔

Reply:
That's exactly why. Users should see the warning and hence it should be `FutureWarning`. `DeprecationWarnings` don't print to the shell and make only sense if this is used from the Python API of rasa which isn't the case for these featurizers.